location/src/main/java/edu/artic/location/FuzedLocationProvider.kt

```diff
@@ -6,35 +6,83 @@ import android.app.Activity
 import android.app.Application
 import android.content.Context
 import android.content.pm.PackageManager
+import android.hardware.Sensor
+import android.hardware.SensorEvent
+import android.hardware.SensorEventListener
+import android.hardware.SensorManager
 import android.location.Location
+import android.location.Location.distanceBetween
 import android.os.Bundle
 import android.os.Handler
 import android.os.Looper
 import android.support.v4.content.ContextCompat
+import android.view.Surface
+import android.view.WindowManager
 import com.google.android.gms.location.LocationCallback
 import com.google.android.gms.location.LocationRequest
 import com.google.android.gms.location.LocationResult
 import com.google.android.gms.location.LocationServices
 import io.reactivex.subjects.BehaviorSubject
 import io.reactivex.subjects.Subject
 
-class FuzedLocationProvider(private val context: Context) : LocationProvider {
+class FuzedLocationProvider(private val context: Context) : LocationProvider, SensorEventListener {
+
+    private val mSensorManager: SensorManager
+    private var mGravity = FloatArray(3)
+    private var mGeomagnetic = FloatArray(3)
+    private var defaultRotation = FloatArray(9)
+    //Rotation after `remapCoordinateSystem` has been invoked to correct for device rotation
+    private var correctedRotation = FloatArray(9)
 
     private var locationRequest: LocationRequest? = null
 
     private val fuzedLocationClient = LocationServices.getFusedLocationProviderClient(context)
 
+    fun configureDeviceAngle() {
+        val windowManager = context .getSystemService(Context.WINDOW_SERVICE) as WindowManager
+        when (windowManager.defaultDisplay.rotation) {
+            Surface.ROTATION_0 ->
+                SensorManager.remapCoordinateSystem(defaultRotation, SensorManager.AXIS_Z,
+                        SensorManager.AXIS_Y, correctedRotation)
+            Surface.ROTATION_90 ->
+                SensorManager.remapCoordinateSystem(defaultRotation, SensorManager.AXIS_Y,
+                        SensorManager.AXIS_MINUS_Z, correctedRotation)
+            Surface.ROTATION_180 ->
+                SensorManager.remapCoordinateSystem(defaultRotation, SensorManager.AXIS_MINUS_Z,
+                        SensorManager.AXIS_MINUS_Y, correctedRotation)
+            Surface.ROTATION_270 ->
+                SensorManager.remapCoordinateSystem(defaultRotation, SensorManager.AXIS_MINUS_Y,
+                        SensorManager.AXIS_Z, correctedRotation)
+        }
+    }
+
     private val locationCallback: LocationCallback = object : LocationCallback() {
         override fun onLocationResult(locationResult: LocationResult?) {
             super.onLocationResult(locationResult)
             locationResult?.let {
```

Comment:
Replace with `locationResult?.lastLocation?.let {` since we don't really need anything else.